`CMake Error at cMake/FreeCAD_Helpers/SetupEigen.cmake:14` encountered while attempting to build FreeCAD 0.20.2 from source tarball

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Forums discussion

_No response_

### Version

0.20 (Release)

### Full version info

```shell
N/A; can't build FreeCAD
```


### Subproject(s) affected?

None

### Issue description

Encountered following CMake error while attempting to build FreeCAD from source tarball

```
CMake Error at cMake/FreeCAD_Helpers/SetupEigen.cmake:14 (if):
  if given arguments:

    "VERSION_LESS" "3.3.1"

  Unknown arguments specified
Call Stack (most recent call first):
  CMakeLists.txt:66 (SetupEigen)

```


### Anything else?

Error occurred while attempting to run `cmake .. -DVTK_DIR=/home/kasm-default-profile/install_files/VTK-9.2.6/build` in FreeCAD build directory, `/home/kasm-default-profile/install_files/FreeCAD-0.20.2/freecad-build` (see https://github.com/tetrabiodistributed/tetra-workspace-image/blob/c71158efc224878186f2b1e320643b6ac7fbb438/playbook.yaml#L587)

Full build error can be found at https://github.com/tetrabiodistributed/tetra-workspace-image/actions/runs/4923028963/jobs/8794397233#step:5:1130

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

```
CMake Error at cMake/FreeCAD_Helpers/SetupEigen.cmake:14 (if):
  if given arguments:

    "VERSION_LESS" "3.3.1"

  Unknown arguments specified
Call Stack (most recent call first):
  CMakeLists.txt:66 (SetupEigen)
```


Can we make CMake give a more explicit warning/error ?

--- a/cMake/FreeCAD_Helpers/SetupEigen.cmake
+++ b/cMake/FreeCAD_Helpers/SetupEigen.cmake
@@ -11,14 +11,14 @@ macro(SetupEigen)
                 "=================\n")
     endif(NOT EIGEN3_FOUND)
 
-    if (${EIGEN3_VERSION} VERSION_LESS "3.3.1")
+    if (EIGEN3_FOUND AND ${EIGEN3_VERSION} VERSION_LESS "3.3.1")
         message(WARNING "Disable module flatmesh because it requires "
                         "minimum Eigen3 version 3.3.1 but version ${EIGEN3_VERSION} was found")
         set (BUILD_FLAT_MESH OFF)
     endif()
 
     # Older versions raise the warning -Wdeprecated-copy with clang10/gcc10
-    if (${EIGEN3_VERSION} VERSION_LESS "3.3.8")
+    if (EIGEN3_FOUND AND ${EIGEN3_VERSION} VERSION_LESS "3.3.8")
         unset(_flag_found CACHE)
         check_cxx_compiler_flag("-Wno-deprecated-copy" _flag_found)
         if (_flag_found)

